crates/interfaces/src/sync.rs

```diff
@@ -53,6 +53,15 @@ pub enum SyncState {
 }
 
 impl SyncState {
+    /// Create a sync state from progress.
+    pub fn from_progress(progress: Option<u64>, downloading: bool) -> Self {
```

Comment:
I feel like this should not be here.

very hard to understand what this does without the context of the pipeline.
can just move to pipeline?